### PR TITLE
Add helm charts and helmfile stages to deploy Klyshko

### PR DESCRIPTION
**Pull request by strieflin**
Signed-off-by: Sven Trieflinger <sven.trieflinger@de.bosch.com>

### CODE REVIEW

File: helmfile.d/0400.klyshko.yaml [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+releases:
+  - name: {{ requiredEnv "RELEASE_NAME" }}-klyshko-operator
+    chart: carbynestack-oci/klyshko-operator
+    version: "0.1.0-basic-implementation"
+    values:
+      - controller:
+          image:
+            registry: {{ env "KLYSHKO_IMAGE_REGISTRY" | default "ghcr.io" }}
+            repository: {{ env "KLYSHKO_IMAGE_REPOSITORY" | default "carbynestack/klyshko-operator-controller" }}
+            tag: {{ env "KLYSHKO_IMAGE_TAG" | default "basic-implementation" }}
```
Comment (sbckr) [inline helmfile.d/0400.klyshko.yaml:28]: has tag to be updated once /carbynestack/klyshko#1 has been merged?

File: helmfile.d/0400.klyshko.yaml [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+            pullPolicy: {{ env "KLYSHKO_IMAGE_PULL_POLICY" | default "IfNotPresent" }}
+      - provisioner:
+          image:
+            registry: {{ env "KLYSHKO_IMAGE_REGISTRY" | default "ghcr.io" }}
+            repository: {{ env "KLYSHKO_IMAGE_REPOSITORY" | default "carbynestack/klyshko-provisioner" }}
+            tag: {{ env "KLYSHKO_IMAGE_TAG" | default "basic-implementation" }}
```
Comment (sbckr) [inline helmfile.d/0400.klyshko.yaml:34]: has tag to be updated once /carbynestack/klyshko#1 has been merged?

File: helmfile.d/0450.klyshko-config.yaml [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+  - name: {{ requiredEnv "RELEASE_NAME" }}-klyshko-config
+    chart: ../charts/klyshko
+    values:
+      - scheduler:
+          enabled: {{ env "IS_MASTER" | default "false" }}
+          concurrency: {{ env "KLYSHKO_CONCURRENCY" | default 2 }}
+          threshold: {{ env "KLYSHKO_THRESHOLD" | default 50000 }}
+          ttlSecondsAfterFinished: {{ env "KLYSHKO_TTL_SECONDS_AFTER_FINISHED" | default 120 }}
+          generator:
+            image: {{ env "KLYSHKO_GENERATOR_IMAGE" | default "ghcr.io/carbynestack/klyshko-mp-spdz:latest" }}
```
Comment (sbckr) [inline helmfile.d/0450.klyshko-config.yaml:24]: tags should be pinned to exact versions.
Has tag to be updated once /carbynestack/klyshko#1 has been merged?

File: charts/klyshko/templates/engine-params-extra.yaml [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+#
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: io.carbynestack.engine.params.extra
+data:
+  {{- $pid := (.Values.playerId | toString) -}}
+  {{- range $k, $v := .Values.engineParams.macKeyShares }}
+  {{- if ne $k $pid }}
+  {{ $k }}_mac_key_share_p: |
```
Comment (sbckr) [inline charts/klyshko/templates/engine-params-extra.yaml:15]: Is there a reason the own mac key share in `io.carbynestack.engine.params.secret` are considered as secret but other parties mac keys are stored in a _ConfigMap_?

File: charts/klyshko/values.yaml [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+scheduler:
+  enabled: false
+  concurrency: 2
+  threshold: 50000
+  ttlSecondsAfterFinished: 120
+  generator:
+    image: ghcr.io/carbynestack/klyshko-mp-spdz:latest
```
Comment (sbckr) [inline charts/klyshko/values.yaml:13]: pin version?

File: charts/klyshko/values.yaml [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+    imagePullPolicy: IfNotPresent
+
+playerCount:
+playerId:
+engineParams:
+  prime:
+  macKeyShares:
+    0:
```
Comment (sbckr) [inline charts/klyshko/values.yaml:21]: Is the engine used limited to 2 players only - why the hard binding?

**Review by sbckr — CHANGES_REQUESTED** [review]
Please note comments in files.

File: helmfile.d/0400.klyshko.yaml (hunk 1 above)
Comment (strieflin) [reply helmfile.d/0400.klyshko.yaml:28]: Short answer: yes. Long answer: absolutely. 😁

File: charts/klyshko/templates/engine-params-extra.yaml (hunk 4 above)
Comment (strieflin) [reply charts/klyshko/templates/engine-params-extra.yaml:15]: That's due to how the Klyshko Integration Interface works. There is no other way to provide additional parameters. The setting is anyway insecure with fake tuple generation, so this doesn't seem to be a problem to me. In the future, one might consider the KII to allow for extra parameters to be a config map or a secret. But that needs additional logic in Klyshko that is out-of-scope for this PR as of my opinion.

File: helmfile.d/0400.klyshko.yaml [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+#
+# Deploys a Carbyne Stack Klyshko Offline Tuple Generation service.
+#
+# References:
+#     https://github.com/carbynestack/klyshko
+#
+releases:
+  - name: {{ requiredEnv "RELEASE_NAME" }}-klyshko
+    chart: carbynestack-oci/klyshko
+    version: "0.1.5"
```
Comment (sbckr) [inline helmfile.d/0400.klyshko.yaml:22]: requirees carbynestack/klyshko#49 to be merged

**Review by sbckr — APPROVED** [review]
LGTM

File: charts/klyshko/values.yaml (hunk 5 above)
Comment (strieflin) [reply charts/klyshko/values.yaml]: Pinned to 0.1.4.

File: helmfile.d/0450.klyshko-config.yaml (hunk 3 above)
Comment (strieflin) [reply helmfile.d/0450.klyshko-config.yaml]: Set to 0.1.4.

File: helmfile.d/0400.klyshko.yaml (hunk 2 above)
Comment (strieflin) [reply helmfile.d/0400.klyshko.yaml]: Updated to 0.1.0.

File: helmfile.d/0400.klyshko.yaml (hunk 1 above)
Comment (strieflin) [reply helmfile.d/0400.klyshko.yaml]: Updated to 0.1.1.

**Pull request merged**